src/object_map.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * Allows creating an object map type with a dynamic key type.
+ *
+ * TypeScript only allows `string` for `K` in `{[key: K]: V}` so we need a utility to bridge
+ * the gap.
+ *
+ * This is an alias for TypeScript’s `Record` type, but the name “record” is confusing given our
+ * Airtable domain model.
+ *
+ * @hidden
+ */
+export type ObjectMap<K extends PropertyKey, V> = {[P in K]: V};
```

Comment:
could we change this to use `string` instead of `PropertyKey`?
```suggestion
export type ObjectMap<K extends string, V> = {[P in K]: V};
```

Projects that use TypeScript's [`keyofStringsOnly` option](https://www.typescriptlang.org/docs/handbook/compiler-options.html) will give errors with this, and we don't currently need the full definition of `PropertyKey` which also includes numbers and symbols.